sigstore/_internal/sct.py

```diff
@@ -163,7 +163,7 @@ def verify_sct(
 
     digitally_signed = _pack_digitally_signed(sct, cert, issuer_key_hash)
 
-    if sct.signature_hash_algorithm == hashes.SHA256():
+    if sct.signature_hash_algorithm.name != hashes.SHA256().name:
```

Comment:
Ugh, instance comparisons 🙂 -- maybe `type(sct.signature_hash_algorithm) is not hashes.SHA256`?

Reply:
```suggestion
    if not isinstance(sct.signature_hash_algorithm,  hashes.SHA256):
```